x/gateway/keeper/msg_server_stake_gateway.go

```diff
@@ -24,46 +26,67 @@ func (k msgServer) StakeGateway(
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
 	logger := k.Logger().With("method", "StakeGateway")
-	logger.Info(fmt.Sprintf("About to stake gateway with msg: %v", msg))
+	logger.Info(fmt.Sprintf("about to stake gateway with msg: %v", msg))
 
 	if err := msg.ValidateBasic(); err != nil {
-		return nil, err
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	// Retrieve the address of the gateway
+	gatewayAddress, err := sdk.AccAddressFromBech32(msg.Address)
+	// NB: This SHOULD NEVER happen because msg.ValidateBasic() validates the address as bech32.
+	if err != nil {
+		// TODO_TECHDEBT(#384): determine whether to continue using cosmos logger for debug level.
+		logger.Info(fmt.Sprintf("could not parse address %q", msg.Address))
```

Comment:
This should either be an error or a warning. Please also log the error in there as well

Reply:
I have a strong and principled opinion here. I'm confident that we've discussed this before but it's been a while so please let me reiterate my thinking on this:

1. In my opinion, error, warn, and even info logs should be reserved for logs which are relevant to the **service operator**; as opposed to the end user, for example. These logs should be important, actionable, or indicative of a status change of the service. 

    The rationale here is that in the absence of such prescriptive usage of the log levels, **logs which are relevant to the service operator are lost in the noise** of irrelevant logs.
3. Debug logs should be used for everything else. In all the cases where I've been switching log levels, these logs are **not intended for the validator operator but instead for protocol developers during development and testing**.
4. Since the cosmos-sdk debug level is practically useless (full of cosmos-sdk internal logs), we chose to use the info level for debug logs on-chain.
5. The long-term solution I was imagining was to add a cosmos-sdk logger backed implementation of the [polylog interface](https://github.com/pokt-network/poktroll/blob/main/pkg/polylog/interface.go#L26) that would shunt the existing debug level logs to a new level (e.g. `cosmos-debug` or something) and give us back a useful debug level.

Reply:
Note that the `[Logging]` section of #384 is outstanding. 

Regarding automating/improving alignment with this approach: I could imagine a CI task which collects all the non debug logs added (plus a few lines of context) in a PR and leaves them in a comment.

Reply:
I've updated the logs that I changed to follow this pattern:

```go
err := examplemodtypes.ErrExamplemodSomeError(...)
logger.Info(fmt.Sprintf("ERROR: %s", err)
return nil, status.Error(codes.InvalidArgument, error.Error())
```